src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((obj) => {
+    const objKey = `${obj[key]}`;
+
+    if (!result[objKey]) {
+      result[objKey] = [obj];
+    } else {
+      result[objKey].push(obj);
+    }
```

Comment:
```suggestion
    if (!result[objKey]) {
      result[objKey] = [];
    } 
    
    result[objKey].push(obj);
```

We can simplify it a bit